packages/integration-tests/src/reactiveHints.test.ts

```diff
@@ -0,0 +1,186 @@
+  });
+
+  it("can do immediate primitive field names", () => {
+    expect(reverseReactiveHint(Author, "firstName")).toEqual([{ entity: Author, fields: ["firstName"], path: [] }]);
+  });
+
+  it("can do parent primitive field names", () => {
+    expect(reverseReactiveHint(Book, { author: ["firstName", "lastName"] })).toEqual([
+      { entity: Book, fields: ["author"], path: [] },
+      { entity: Author, fields: ["firstName", "lastName"], path: ["books"] },
+    ]);
+  });
+
+  it("can do grand-parent primitive field names", () => {
+    expect(reverseReactiveHint(BookReview, { book: { author: ["firstName", "lastName"] } })).toEqual([
+      { entity: BookReview, fields: ["book"], path: [] },
+      { entity: Book, fields: ["author"], path: ["reviews"] },
+      { entity: Author, fields: ["firstName", "lastName"], path: ["books", "reviews"] },
+    ]);
+  });
+
+  it("can do parent and grand-parent primitive field names", () => {
+    expect(reverseReactiveHint(BookReview, { book: { title: {}, author: ["firstName", "lastName"] } })).toEqual([
+      { entity: BookReview, fields: ["book"], path: [] },
+      { entity: Book, fields: ["title", "author"], path: ["reviews"] },
+      { entity: Author, fields: ["firstName", "lastName"], path: ["books", "reviews"] },
+    ]);
+  });
+
+  it("can do child o2m with primitive field names", () => {
+    expect(reverseReactiveHint(Author, { books: "title" })).toEqual([
+      {
+        entity: Book,
+        fields: ["author", "title"],
+        path: ["author"],
+      },
+    ]);
+  });
+
+  it("can do child o2m with w/o any ", () => {
+    expect(reverseReactiveHint(Author, "books")).toEqual([
+      {
+        entity: Book,
+        fields: ["author"],
+        path: ["author"],
+      },
+    ]);
+  });
+
+  it("can do child o2o with primitive field names", () => {
+    expect(reverseReactiveHint(Author, { image: "fileName" })).toEqual([
+      {
+        entity: Image,
+        fields: ["author", "fileName"],
+        path: ["author"],
+      },
+    ]);
+  });
+
+  describe("convertToLoadHint", () => {
+    it("works with child o2o and primitive field names", () => {
+      expect(convertToLoadHint(getMetadata(Author), { image: "fileName" })).toEqual({ image: {} });
+    });
+
+    it("works with child o2m and primitive field names", () => {
+      expect(convertToLoadHint(getMetadata(Author), { books: "title" })).toEqual({ books: {} });
+    });
+
+    it("works with parent m2o and primitive field names", () => {
+      expect(convertToLoadHint(getMetadata(Author), { publisher: "name" })).toStrictEqual({ publisher: {} });
+      expect(convertToLoadHint(getMetadata(Book), { author: "firstName" })).toStrictEqual({ author: {} });
+      expect(convertToLoadHint(getMetadata(BookReview), { book: { author: "firstName" } })).toStrictEqual({
+        book: { author: {} },
+      });
+    });
+  });
+
+  describe("typings", () => {
```

Comment:
curious: Since there are no `it` blocks, or expects, I'm guessing these are ensuring that the code passes type checking?